infrastructure/src/main/java/org/corfudb/infrastructure/ManagementAgent.java

```diff
@@ -396,6 +394,15 @@ private boolean runRecoveryReconfiguration() {
         getCorfuRuntime().invalidateLayout();
         Layout clusterLayout = getCorfuRuntime().getLayoutView().getLayout();
 
+        // Trigger sequencer recovery only if this node was successful in committing the recovery
+        // layout. If not, then we can assume that the sequencer is already bootstrapped.
+        // NOTE: This is an async task and the failure of this task does not affect the recovery.
+        if (recoveryReconfigurationResult) {
+            // Save the latest management layout.
+            serverContext.saveManagementLayout(getCorfuRuntime().getLayoutView().getLayout());
```

Comment:
Line 395 `getCorfuRuntime().getLayoutView().getLayout()` why we get layout twice? Is it possible to reuse clusterLayout?

Reply:
Done.